src/custom/hooks/useRefetchPriceCallback.tsx

```diff
@@ -217,6 +207,38 @@ export function useRefetchQuoteCallback() {
           error: quoteError,
         })
       }
+
+      const { sellToken, buyToken, chainId } = quoteData
+      // Start action: Either new quote or refreshing quote
+      if (isPriceRefresh) {
+        // Refresh the quote
+        refreshQuote({ sellToken, chainId })
+      } else {
+        // Get new quote
+        getNewQuote(quoteParams)
+      }
+
+      // Init get quote methods params
+      const bestQuoteParams = { ...params, strategy: priceStrategy }
+      const fastQuoteParams = { quoteParams: { ...quoteParams, priceQuality: 'fast' } }
+
+      // Register get best and fast quote methods on window
+      registerOnWindow({
+        getBestQuote: async () => getBestQuoteResolveOnlyLastCall(bestQuoteParams),
+        getFastQuote: async () => getFastQuoteResolveOnlyLastCall(fastQuoteParams),
+      })
+
+      // Get the fast quote
+      if (!isPriceRefresh) {
+        getFastQuoteResolveOnlyLastCall(fastQuoteParams)
+          .then((res) => handleResponse(res, false))
+          .catch(handleError)
+      }
+
+      // Get the best quote
+      getBestQuoteResolveOnlyLastCall(bestQuoteParams)
+        .then((res) => handleResponse(res, true))
+        .catch(handleError)
```

Comment:
Can there be a race condition? 

`getFastQuoteResolveOnlyLastCall` is sent in parallel to `getBestQuoteResolveOnlyLastCall`. What happens if `getBestQuoteResolveOnlyLastCall` finish before the fast one? Wouldn't we update using the wrong price in this case?

Reply:
It would, the solution for this is in latest commit and I tested it in a way by using setTimeout on fast price update action. And the update will check inside of update action code, if there is already a quote price amount, which means that the best quote request was already done and if the current action params are bestQuote update or not.